staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -282,66 +296,59 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 					return cmdutil.UsageErrorf(cmd, "couldn't find port via --port flag or introspection")
 				}
 			case 1:
-				params["port"] = ports[0]
+				o.Port = ports[0]
 			default:
-				params["ports"] = strings.Join(ports, ",")
+				o.Ports = strings.Join(ports, ",")
 			}
 		}
 
 		// Always try to derive protocols from the exposed object, may use
 		// different protocols for different ports.
-		if _, found := params["protocol"]; found {
-			protocolsMap, err := o.ProtocolsForObject(info.Object)
-			if err != nil {
-				return cmdutil.UsageErrorf(cmd, "couldn't find protocol via introspection: %v", err)
-			}
-			if protocols := generate.MakeProtocols(protocolsMap); !generate.IsZero(protocols) {
-				params["protocols"] = protocols
-			}
+		protocolsMap, err := o.ProtocolsForObject(info.Object)
+		if err != nil {
+			return cmdutil.UsageErrorf(cmd, "couldn't find protocol via introspection: %v", err)
```

Comment:
I'd prefer we switch all internal errors to just return error, so `fmt.Errorf("couldn't find protocol via introspection: %v", err)` in the rest of the file as well. This allows consumers of the command to wrap the error and respond accordingly. Whereas `cmdutil.UsafeErrorf` adds unnecessarily kubectl specific information, ie. `See 'kubectl expose -h' for help and examples`, etc, which are already nicely handled in the cobra's `Run` command.

Reply:
Fixed